packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1782,9 +1782,9 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
       final ClipboardData? data = await Clipboard.getData(Clipboard.kTextPlain);
       if (data != null) {
         value = TextEditingValue(
-          text: selection.textBefore(text) + data.text! + selection.textAfter(text),
```

Comment:
for selection is invalid we will append it to the end, I think this is probably fine, but do we know what the native behavior is?

Reply:
I am not sure we can set the selection invalid on the native platform cc @LongCatIsLooong @justinmc

Reply:
Yes (at least for iOS and android): 
- iOS https://developer.apple.com/documentation/uikit/uitextinput/1614541-selectedtextrange can be `nil`
- Android there can be no span at all, and you can remove the selection from your text using https://developer.android.com/reference/android/text/Selection?hl=en#removeSelection(android.text.Spannable)